aten/src/ATen/native/cuda/EmbeddingBag.cu

```diff
@@ -247,6 +247,25 @@ Tensor embedding_bag_backward_cuda_max(const Tensor &grad,
 }
 }
 
+// Assumes all input tensors are contiguous.
+// See NOTE [ embedding_bag Native Functions ] in native_functions.yaml for details
+std::tuple<Tensor, Tensor, Tensor, Tensor>
+_embedding_bag_forward_only_cuda(const Tensor &weight, const Tensor &indices,
+                   const Tensor &offsets, const bool scale_grad_by_freq,
+                   const int64_t mode, bool sparse,
+                   const Tensor& per_sample_weights,
+                   bool include_last_offset) {
+  return _embedding_bag_cuda(
```

Comment:
Do you happen to know why cuda version doesn't depend on requires_grad information at all?

Reply:
CUDA version always computes [bag_size](https://github.com/pytorch/pytorch/blob/63e5a53b8cc4d0329f39cec153eeed8119d1fe04/aten/src/ATen/native/cuda/EmbeddingBag.cu#L62)

CPU version sometimes computes bag_size due to an [optimization](https://github.com/pytorch/pytorch/commit/8d35b6cec7146f0cffded101b806eb4dc3d47daf)

In other words, GPU version didn't have same problem due to lack of this optimization